File: 202102655/1018.py
N, M = map(int, input().split())
chess = []
for _ in range(N):
    chess.append(list(input()))
min_cnt = 10000000
for i in range(N-8+1):
    for j in range(M-8+1):
        check_wb = False
        cnt = 0
        cnt_reverse = 0
        if chess[i][j] == 'B':
            check_wb = True
        for x in range(8):
            for y in range(8):
                if (chess[i+x][j+y] == 'W' and check_wb) or (chess[i+x][j+y] == 'B' and not check_wb):
                    cnt += 1
                elif (chess[i+x][j+y] == 'W' and not check_wb) or (chess[i+x][j+y] == 'B' and check_wb):
                    cnt_reverse += 1
                if y != 7:
                    check_wb = not check_wb
        min_cnt = min(min_cnt, cnt_reverse)
        min_cnt = min(min_cnt, cnt)
print(min_cnt)

File: 202402720/1018.go
package main

import (
	"fmt"
	"strings"
)

var (
	tile [][]string

	firstWTile = [][]string{
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
	}

	firstBTile = [][]string{
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
		{"B", "W", "B", "W", "B", "W", "B", "W"},
		{"W", "B", "W", "B", "W", "B", "W", "B"},
	}
)

func getRemaningCount(x int, y int) int {
	var firstCount int
	var secondCount int

	for i := x; i < x+8; i++ {
		for j := y; j < y+8; j++ {
			if tile[i][j] != firstWTile[i-x][j-y] {
				firstCount++
			}
			if tile[i][j] != firstBTile[i-x][j-y] {
				secondCount++
			}
		}
	}

	if firstCount < secondCount {
		return firstCount
	} else {
		return secondCount
	}
}

func main() {
	var N, M int
	fmt.Scan(&N, &M)

	tile = make([][]string, N)
	for i := range tile {
		tile[i] = make([]string, M)
	}

	for i := 0; i < N; i++ {
		var input string
		fmt.Scan(&input)

		for j, v := range strings.Split(input, "") {
			tile[i][j] = v
		}
	}

	min := 1 << 31

	for i := 0; i <= N-8; i++ {
		for j := 0; j <= M-8; j++ {
			v := getRemaningCount(i, j)
			if v < min {
				min = v
			}
		}
	}

	fmt.Println(min)
}
